test/wallet-http-test.js

```diff
@@ -0,0 +1,558 @@
+/* eslint-env mocha */
+/* eslint prefer-arrow-callback: "off" */
+
+'use strict';
+
+const assert = require('./util/assert');
+const {rimraf, sleep} = require('./util/common');
+
+const {
+  initFullNode,
+  initSPVNode,
+  initNodeClient,
+  initWalletClient,
+  initWallet,
+  generateInitialBlocks,
+  generateTxs
+} = require('./util/regtest');
+
+const testPrefix = '/tmp/bcoin-fullnode';
+const spvTestPrefix = '/tmp/bcoin-spvnode';
+const genesisTime = 1534965859;
+const genesisDate = new Date(genesisTime * 1000);
+
+const ports = {
+  full: {
+    p2p: 49331,
+    node: 49332,
+    wallet: 49333
+  },
+  spv: {
+    p2p: 49431,
+    node: 49432,
+    wallet: 49433
+  }
+};
+
+describe('Wallet TX HTTP Pagination', function() {
+  this.timeout(60000);
+
+  let node, spvnode, wallet = null;
+  let nclient, wclient, spvwclient = null;
+  let coinbase = null;
+  let unconfirmedTime = null;
+  let unconfirmedDate = null;
+
+  before(async () => {
+    await rimraf(testPrefix);
+    await rimraf(spvTestPrefix);
+
+    node = await initFullNode({
+      ports,
+      prefix: testPrefix,
+      logLevel: 'none'
+    });
+
+    spvnode = await initSPVNode({
+      ports,
+      prefix: spvTestPrefix,
+      logLevel: 'none'
+    });
+
+    nclient = await initNodeClient({ports: ports.full});
+    wclient = await initWalletClient({ports: ports.full});
+    spvwclient = await initWalletClient({ports: ports.spv});
+    wallet = await initWallet(wclient);
+    await initWallet(spvwclient);
+
+    await wclient.execute('selectwallet', ['test']);
+    coinbase = await wclient.execute('getnewaddress', ['blue']);
+
+    await spvwclient.execute('selectwallet', ['test']);
+
+    await generateInitialBlocks({
+      nclient,
+      wclient,
+      spvwclient,
+      coinbase,
+      genesisTime,
+      blocks: 125
+    });
+
+    // Generate unconfirmed transactions for the
+    // fullnode and spv wallet
+    unconfirmedTime = Math.floor(Date.now() / 1000);
+    unconfirmedDate = new Date(unconfirmedTime * 1000);
+
+    // TODO remove this
+    await sleep(5000);
+
+    await generateTxs({
+      wclient,
+      spvwclient,
+      count: 19,
+      amount: 0.0001,
+      gap: 6,
+      sleep: 2000
+    });
+
+    // TODO remove this. It's necessary to wait because
```

Comment:
Would it be possible to set up a websocket listener for the `tx` event instead of sleeping?

Reply:
Can probably use some of the testing utils added in another PR, `await forValue()` and etc.

Reply:
Added.